index.ts

```diff
@@ -86,15 +86,43 @@ export default function scss(options: CSSPluginOptions = {}): Plugin {
                 prev: string,
                 done: ImporterDoneCallback
               ): ImporterReturnType | void => {
-                if (url.startsWith('.')) return null
-
+                // If the import begins with '.', it's local...abort. If it
+                // begins with 'url', 'http', or 'https', it's a remote import,
+                // again abort. Note: we only check for 'http' because both
+                // 'http' and 'https' begin with
+                if (
+                  url.startsWith('.') ||
+                  url.startsWith('url') ||
+                  url.startsWith('http')
+                ) {
+                  return null
+                }
+
+                // If the url begins with '~' remove it. The '~' character is a
+                // standard in the sass community when working in JavaScript
+                // because sass-loader used it as an indicator that the import
+                // was a node_modules import (and still does, I think).
                 const cleanUrl = url.startsWith('~')
                   ? url.replace('~', '')
                   : url
+
+                // Have Node resolve the path to the file we're requesting.
+                // When using a plain node_modules project, this works as
+                // expected. When using Yarn PnP, Yarn will unplug the package,
+                // and give the path to the file. Unplugging a file simply
+                // means unzipping it since most packages aren't equipped to
+                // work with zip archives directly. Also, the 'paths' option
+                // tells Yarn what file is requesting this file, so PnP
+                // integrity remains intact. For Node, paths serves as an
+                // additional set of paths to look for the import in.
```

Comment:
Yarn doesn't unplug on demand, it's done on install if required due to [native files](https://github.com/yarnpkg/berry/blob/6e445c255282aa2cd619d8d91c9c4d921c1de6f2/packages/plugin-pnp/sources/jsInstallUtils.ts#L65-L74), postinstalls, or when explicitly requested.
All packages that use the `fs` module can work with zip files as PnP patches it to support reading from them.

> For Node, paths serves as an additional set of paths to look for the import in

For both PnP and Node it's to tell it where the resolution should start, so it's not "additional"
https://nodejs.org/dist/latest-v16.x/docs/api/modules.html#modules_require_resolve_request_options

Reply:
Fair enough, that was a misunderstanding on my part. PnP is such a massive departure from what I'm used to that I am still coming to grips with it (not that that's a bad thing, this kinda needed to happen for a while). As for the `fs` module support is that due to Yarn patching it, or is that just native node?

Also, with regards to the resolution, that was really just overly technical documentation. The way it is phrased makes it sound as if it's adding new resolution paths, but I'll update the comment. Thanks for clearing that up.

Reply:
Does this phrasing capture a bit more accurately what's going on?

```ts
// Have Node resolve the path to the file we're requesting.
// When using a plain node_modules project, this works as
// expected. When using Yarn PnP, Yarn will resolve an absolute
// path to the file and the sass compiler will access that file
// directly from the zip archive it's contained in. The 'paths'
// option simply tells Node and Yarn where resolution should
// start (i.e. who is requesting this file).
```

Reply:
I mean technically, both Node and Yarn PnP are doing the same thing, so it's a bit misleading. They're both just getting the absolute path to a file. It's just in the case of Node, it's the runtime doing it and with Yarn PnP, it's Yarn doing it.

Reply:
Alternative verbiage, I'll let you decide which is more accurate and concise.

```ts
// Have Node or PnP resolve a path to the file we're
// requesting. In the case of a standard node_modules project
// Node will handle the resolution. For a PnP project, the
// package manager will handle the resolution. Both will
// return an absolute path to the file, which will be passed to
// the sass compiler, so it can get the file in question.
//
// If a file cannot be located, this will generate an error.
```

I also don't specifically mention Yarn, since it isn't the only PnP based package manager.

Reply:
Sorry for the late response, the way you ended up wording it looks great

> As for the fs module support is that due to Yarn patching it, or is that just native node?

The PnP implementation patches it
https://github.com/yarnpkg/berry/blob/d2e592a74e1de6fcf9e2e97a5d9712cf7eba159c/packages/yarnpkg-pnp/sources/loader/_entryPoint.ts#L26-L37